src/compiler/transformers/declarations.ts

```diff
@@ -638,7 +638,7 @@ export function transformDeclarations(context: TransformationContext) {
             case SyntaxKind.PropertyDeclaration:
             case SyntaxKind.BindingElement:
             case SyntaxKind.VariableDeclaration:
-                typeNode = resolver.createTypeOfDeclaration(node, enclosingDeclaration, declarationEmitNodeBuilderFlags, symbolTracker, shouldAddImplicitUndefined);
+                typeNode = resolver.createTypeOfDeclaration(node, enclosingDeclaration, declarationEmitNodeBuilderFlags, symbolTracker);

```

Comment:
Do we still need `shouldAddImplicitUndefined` at this point? I did a diff locally forcing it to false above and it was... almost clean. So I think the answer is "yes" until we fix parameter properties and hopefully our goofy rules about optionalish/requiredish parameters...

Reply:
> Do we still need shouldAddImplicitUndefined at this point?

Ultimately, I would like to remove it and the `EmitResolver` API backing it - for me, that was what motivated basically this whole stack of changes (quite literally: I removed it, and then started seeing what needed to be done to minimize the diffs). There's a bit more work to be done to minimize that diff, though - mostly, from what I can tell, around handling errors and duplicated declarations (with structurally-identical-but-identity-unique types) in the node builder more like the top-level declaration emitter does today.

Reply:
> So I think the answer is "yes" until we fix parameter properties and hopefully our goofy rules about optionalish/requiredish parameters...

Ah, you want to lock it to `true`, not `false`, that way it always falls back to the logic within `createTypeOfDeclaration` (which handles optional properties), rather than the declaration emitter's `visitDeclarationSubtree` (which doesn't) - that'll be more representative of what needs to be fixed in the node builder to remove the API.

Reply:
> I meant false in the code above (I couldn't highlight it in the diff, thanks GitHub), such that type && !shouldAddImplicitUndefined is true and we just copy the nodes (so, more reuse).

Yeah, exactly, at no point have I changed this case in this or any recent PR, however. The `visitDeclarationSubtree` codepath _always_ copies nodes, without regard for type information that may indicate otherwise. That's right enough most of the time, without considering the `shouldAddImplicitUndefined` carve-out. What we care about, IMO, is if the fallback `createTypeOfDeclaration` calls _also_ always copy the input nodes when they can - because if they do, the entire 
```ts
if (type && !shouldAddImplicitUndefined) {
    return visitNode(type, visitDeclarationSubtree, context);
}
```
branch can be removed, as `createTypeOfDeclaration` will internally have all the same logic (and it needs to anyway, in order to handle copying subtrees of input nodes within the nodes it generates). After this change, the only issues blocking that, AFAIK, are that `createTypeOfDeclaration` over-invalidates in the presence of an error type/structurally identical types across multiple declarations, and discards input nodes it could probably safely reuse.